src/utils/condition-validation.ts

```diff
@@ -42,4 +42,21 @@ const checkConditionValidity = (condition: EntityCondition | AccountCondition):
   }
 };
 
+export const validateExpiryTimeDate = (xprtnDtTm: string): string => {
+  if (xprtnDtTm) {
+    const expirationDate = new Date(xprtnDtTm);
+    if (isNaN(expirationDate.getTime())) {
+      throw Error('Error: Expiration date provided was invalid.');
+    }
+    if (expirationDate.getTime() < Date.now()) {
+      throw Error('Error: Expiration date must be after current time and date');
+    }
+  }
+
+  if (!xprtnDtTm) {
+    return new Date().toISOString();
+  }
+  return xprtnDtTm;
+};
+
 export default checkConditionValidity;
```

Comment:
1. This function is trying to do too much on it's own. I suggest you break it up into easier managed pieces. 
Here's a starting snippet
```
export const convertDateToISO = (dateStr: string): string | null {
  if (!isDateValid(dateStr)) {
    throw Error('Error: Date provided was invalid.')
  }

  const date = new Date(dateStr);
  return date.toISOString();
}

const isDateValid = (dateStr: string): boolean {
  const date = new Date(dateStr);
  return !isNaN(date.getTime());
}

export const hasDateExpired = (date: Date): boolean {
  const now = new Date();
  return date < now;
}
```

2. Also you have a default export and now non default exports. I suggest you keep it consistent.